src/EpochIOGroup.cpp

```diff
@@ -0,0 +1,397 @@
+                            "not valid for EpochIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        int signal_type = signal_name_to_type(signal_name);
+        std::function<double(const std::vector<double> &)> result = Agg::max;
+        switch (signal_type) {
+            case M_SIGNAL_EPOCH_COUNT:
+                result = Agg::min;
+                break;
+            case M_SIGNAL_EPOCH_RUNTIME:
+            case M_SIGNAL_EPOCH_RUNTIME_NETWORK:
+            case M_SIGNAL_EPOCH_RUNTIME_IGNORE:
+                break;
+            default:
+#ifdef GEOPM_DEBUG
+                throw Exception("EpochIOGroup::sample(): invalid signal type",
+                                GEOPM_ERROR_LOGIC, __FILE__, __LINE__);
+#endif
+                break;
+        }
+        return result;
+    }
+
+    std::function<std::string(double)> EpochIOGroup::format_function(const std::string &signal_name) const
+    {
+        if (!is_valid_signal(signal_name)) {
+            throw Exception("EpochIOGroup::format_function(): " + signal_name +
+                            " not valid for EpochIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        int signal_type = signal_name_to_type(signal_name);
+        std::function<std::string(double)> result = string_format_double;
+        switch (signal_type) {
+            case M_SIGNAL_EPOCH_COUNT:
+                result = string_format_integer;
+                break;
+            case M_SIGNAL_EPOCH_RUNTIME:
+            case M_SIGNAL_EPOCH_RUNTIME_NETWORK:
+            case M_SIGNAL_EPOCH_RUNTIME_IGNORE:
+                break;
+            default:
+#ifdef GEOPM_DEBUG
+                throw Exception("EpochIOGroup::sample(): invalid signal type",
+                                GEOPM_ERROR_LOGIC, __FILE__, __LINE__);
+#endif
+                break;
+        }
+        return result;
+    }
+
+    std::string EpochIOGroup::signal_description(const std::string &signal_name) const
+    {
+        if (!is_valid_signal(signal_name)) {
+            throw Exception("EpochIOGroup::signal_description(): " + signal_name +
+                            " not valid for EpochIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        int signal_type = signal_name_to_type(signal_name);
+        std::string result;
+        switch (signal_type) {
+            case M_SIGNAL_EPOCH_COUNT:
+                result = "Number of epoch events sampled from the process on the given CPU";
+                break;
+            case M_SIGNAL_EPOCH_RUNTIME:
+                result = "Last runtime between epoch events sampled from the process on the given CPU";
+                break;
+            case M_SIGNAL_EPOCH_RUNTIME_NETWORK:
+                result = "Portion of the last epoch with the network hint sampled from the process on the given CPU";
```

Comment:
Portion of last runtime between epoch event where the application provided the network hint.